cartography/intel/aws/iam.py

```diff
@@ -270,70 +294,66 @@ def load_group_memberships(neo4j_session, group_memberships, aws_update_tag):
             )
 
 
-def _find_roles_assumable_in_policy(policy_data):
-    ret = []
-    statements = policy_data["PolicyDocument"]["Statement"]
-    if isinstance(statements, dict):
-        statements = [statements]
-    for statement in statements:
-        parsed_statement = policyuniverse.statement.Statement(statement)
-        if parsed_statement.effect == 'Allow' and 'sts:assumerole' in parsed_statement.actions_expanded:
-            ret.extend(list(parsed_statement.resources))
-    return ret
-
-
 @timeit
-def load_group_policies(neo4j_session, group_policies, aws_update_tag):
-    ingest_policies_assume_role = """
-    MATCH (group:AWSGroup{arn: {GroupArn}})
-    WITH group
-    MERGE (role:AWSRole{arn: {RoleArn}})
-    ON CREATE SET role.firstseen = timestamp()
-    SET role.lastupdated = {aws_update_tag}
-    WITH role, group
-    MERGE (group)-[r:STS_ASSUMEROLE_ALLOW]->(role)
-    ON CREATE SET r.firstseen = timestamp()
-    SET r.lastupdated = {aws_update_tag}
+def get_policies_for_principal(neo4j_session, principal_arn):
+    get_policy_query = """
+    MATCH
+    (principal:AWSPrincipal{arn:{Arn}})-[:POLICY]->
+    (policy:AWSPolicy)-[:STATEMENT]->
+    (statements:AWSPolicyStatement)
+    RETURN
+    DISTINCT policy.id AS policy_id,
+    COLLECT(DISTINCT statements) AS statements
     """
-
-    for group_arn, policies in group_policies.items():
-        for policy_name, policy_data in policies.items():
-            for role_arn in _find_roles_assumable_in_policy(policy_data):
-                # TODO resource ARNs may contain wildcards, e.g. arn:aws:iam::*:role/admin --
-                # TODO policyuniverse can't expand resource wildcards so further thought is needed here
-                neo4j_session.run(
-                    ingest_policies_assume_role,
-                    GroupArn=group_arn,
-                    RoleArn=role_arn,
-                    aws_update_tag=aws_update_tag,
-                )
+    results = neo4j_session.run(
+        get_policy_query,
+        Arn=principal_arn,
+    )
+    policies = {r["policy_id"]: parse_statement_node_group(r["statements"]) for r in results}
+    return policies
 
 
 @timeit
-def load_role_policies(neo4j_session, role_policies, aws_update_tag):
+def sync_assume_role(neo4j_session, current_aws_account_id, aws_update_tag, common_job_parameters):
```

Comment:
I would rather keep the name what it is modifying in the graph then how it got the data. Analyze doesn't let me know its making sync changes.

Reply:
How about `sync_assumerole_relationships()` since it creates `STS_ASSUMEROLE_ALLOW` relationships?

Reply:
Oh, also can we add a comment saying that this function *must* be run after users, groups, roles, and group_memberships have been synced?

I initially didn't want to use the `sync` verb because this relies on other IAM nodes already being in the graph and the others don't. I won't block on the verb name, but I think there's a better way to communicate this.

Reply:
Done